spec/environment_spec.rb

```diff
@@ -551,4 +551,469 @@ def new_content(id)
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentA['id']}.1")).to be false
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentC['id']}.1")).to be false
+
+    envB = @org_admin.create_environment(@owner['key'], 'envB', "envB", "For test systems only.")
+    job = @org_admin.promote_content(@env['id'],[{:contentId => contentD['id'],}])
+    wait_for_job(job['id'], 15)
+    consumer_cp.update_consumer({:environments => [{:name => @env.name}, {:name => envB.name}]})
+    ent = consumer_cp.list_entitlements()[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentA['id']}.1")).to be false
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentC['id']}.1")).to be false
+  end
+
+  it 'should re-gen consumer entitlement cert when environment priority are reversed' do
+    envB = @org_admin.create_environment(@owner['key'], 'envB', "envB", "For test systems only.")
+    consumer = @org_admin.register(random_string('testsystem'), :system, nil, {}, nil, nil, [], [],
+      [{ 'id' => @env['id']}, { 'id' => envB['id']}])
+
+    expect(consumer['environments']).not_to be_nil
+
+    consumer_cp = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+    product = create_product
+    contentA = create_content
+    contentB = create_content
+    contentC = create_content
+    @cp.add_content_to_product(@owner['key'], product['id'], contentA['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentB['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentC['id'])
+    job = @org_admin.promote_content(@env['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
+    job = @org_admin.promote_content(envB['id'],[{:contentId => contentC['id'],}])
+    wait_for_job(job['id'], 15)
+    pool = @cp.create_pool(@owner['key'], product['id'])
+    ent = consumer_cp.consume_pool(pool['id'], {:quantity => 1})[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentC['id']}.1")).to be true
+
+    consumer_cp.update_consumer({:environments => [{:name => envB.name}, {:name => @env.name}]})
+    ent = consumer_cp.list_entitlements()[0]
```

Comment:
All the spec test names are given names such as 'should re-gen / should NOT regen', but they aren't actually asserting whether the entitlement serial changed or not. For the ones that happen to have obvious changes in content (like content added/removed) that is not a problem, but for tests like this one, it is, so let's add this check to all of the tests.